src/kfactory/kcell.py

```diff
@@ -1586,21 +1688,9 @@ def __getitem__(self, key: int | str | None) -> Port:
         """Returns port from instance."""
         return self.ports[key]
 
-    @property
-    def settings(self) -> KCellSettings:
-        """Settings dictionary.
-
-        Set by the [@cell][kfactory.kcell.KCLayout.cell] decorator.
-        """
-        return self._settings
-
-    @property
-    def settings_units(self) -> KCellSettingsUnits:
-        """Dictionary containing the units of the settings.
-
-        Set by the [@cell][kfactory.kcell.KCLayout.cell] decorator.
-        """
-        return self._settings_units
+    def __hash__(self) -> int:
+        """Hash the KCell."""
+        return hash(self._kdb_cell)
```

Comment:
Here I would just return `hash((self.kcl.library.name(), self._kdb_cell.cell_index())`

I believe if you just hash the cell, you are taking the literal address of it. Which afaik is not unique, you can have multiple klayout cells pointing to the same cell, but have a different python wrapper.